src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -759,8 +776,58 @@ protected override bool TryParseRequest(ReadResult result, out bool endConnectio
     }
 
 #pragma warning disable CS0618 // Type or member is obsolete
-    private void DetectHttp2Preface(ReadOnlySequence<byte> requestData, BadHttpRequestException ex)
+    private void OnBadRequest(ReadOnlySequence<byte> requestData, BadHttpRequestException ex)
 #pragma warning restore CS0618 // Type or member is obsolete
+    {
+        var reason = ex.Reason;
+
+        // Some code shared between HTTP versions throws errors. For example, HttpRequestHeaders collection
+        // throws when an invalid content length is set.
+        // Only want to set a reasons for HTTP/1.1 connection, so set end reason by catching the exception here.
+        switch (reason)
```

Comment:
@halter73 @Tratcher It would be good to get your thoughts on whether new connection end reasons should be added based on available `RequestRejectionReason` values thrown when parsing a 1.1 request.

Reply:
Maybe for the more common ones that aren't just buggy clients like
UnexpectedEndOfRequestContent
RequestBodyTooLarge
RequestBodyTimeout

Reply:
I agree with the three listed above. TlsOverHttpError, HeadersExceedMaxTotalSize and to a lesser extent TooManyHeaders would probably also be good. The more esoteric ones like MultipleContentLengths and BadChunkSizeData probably aren't worth it.

Reply:
I added TlsOverHttpError, MaxRequestBodySizeExceeded and UnexpectedEndOfRequestContent.

I looked through the changes and I wonder if UnexpectedEndOfRequestContent is correct. Does it end the connection? I think it just ends the request.